Feature/comic list

Merihel opened the pull request:
Pull request for the comic list feature

Commit:
[FEAT] Working comic list with Adapter and ViewHolder (RecyclerView)

- Added ViewHolder that holds data of the comic's card
- Added RecyclerView in ComicListActivity (Main activity)
- Added background card for comics

ronzierLoic committed:
Merge branch 'master' into feature/comic-list

Commit:
[FIX] Remove drawable/deadpool

Deleted "deadpool.jpg" from local drawable, and it's reference in recycler_item_comic layout that caused the build to fail (unknown reference)

Commit:
Merge remote-tracking branch 'origin/feature/comic-list' into feature/comic-list

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/activity/ComicListActivity.java:
```diff
@@ -0,0 +1,85 @@
+
+public class ComicListActivity extends AppCompatActivity {
+
+    private RecyclerView mRecyclerView;
```
Je pense qu'il faudrait changer le nom de la variable en `comicRecyclerView` pour mieux comprendre à quoi il correspond

Merihel replied:
✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/activity/ComicListActivity.java:
```diff
@@ -0,0 +1,85 @@
+public class ComicListActivity extends AppCompatActivity {
+
+    private RecyclerView mRecyclerView;
+    private RecyclerView.Adapter mAdapter;
```
Je pense qu'il faudrait changer le nom de la variable en `recyclerComicListAdapter` pour mieux comprendre à quoi il correspond et son type c'est plutôt `RecyclerComicsListAdapter`

Merihel replied:
✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/activity/ComicListActivity.java:
```diff
@@ -0,0 +1,85 @@
+
+    private RecyclerView mRecyclerView;
+    private RecyclerView.Adapter mAdapter;
+    private RecyclerView.LayoutManager mLayoutManager;
```
Je ne pense pas que cette variable soit utile on l'utilise que une seul fois pour set le layoutManager du recycler view je pense qu'il faudrais plutot l'appeler directement dans le recyclerView comme ceci : `mRecyclerView.setLayoutManager(new LinearLayoutManager(this))`

Merihel replied:
✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/activity/ComicListActivity.java:
```diff
@@ -0,0 +1,85 @@
+        this.setTitle("Liste de comics");
+
+        mRecyclerView = (RecyclerView) findViewById(R.id.comics_list_rv);
+        mRecyclerView.setHasFixedSize(true); //Pas de changement de size du layout -> fixe pour les perfs
```
A quoi sert setHasFixedSize ?

Merihel replied:
C'était à propos d'un cas particulier où le RecyclerView change de size par rapport au parent, je te cache pas que j'ai un peu mis ça au hasard mais si on l'enlève ça devrais pas changer grand chose. Ca ira avec le reste. 👍

ronzierLoic replied:
Okey, bah enleve le si il rajoute rien de spécial ;)

Merihel replied:
Fait ;) ✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/activity/ComicListActivity.java:
```diff
@@ -0,0 +1,85 @@
+        mRecyclerView.setHasFixedSize(true); //Pas de changement de size du layout -> fixe pour les perfs
+
+        mLayoutManager = new LinearLayoutManager(this);
+        mRecyclerView.setLayoutManager(mLayoutManager);
```
Je pense plutôt qu'on devrais set le LayoutManager comme ceci :  `mRecyclerView.setLayoutManager(new LinearLayoutManager(this))`

Merihel replied:
✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/activity/ComicListActivity.java:
```diff
@@ -0,0 +1,85 @@
+        comicList.add(comic3);
+        comicList.add(comic4);
+        comicList.add(comic5);
+        comicList.add(comic6);
```
Il faudrait enlever les test en brut maintenant que l'on sait que ça fonction correctement

Merihel replied:
✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/adapter/RecyclerComicsListAdapter.java:
```diff
@@ -1,4 +1,59 @@
+import java.util.Date;
+import java.util.List;
+
+public class RecyclerComicsListAdapter extends RecyclerView.Adapter<ComicsListHolder> {
```
Peut etre changer le nom de cette classe ou celle de l'activity car une fois il y a un `s` a comics une fois nan :/

Merihel replied:
✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/adapter/RecyclerComicsListAdapter.java:
```diff
@@ -1,4 +1,59 @@
+    private List<ResultsItem> itemList;
+
+    //Constructeur à adapter selon les données
+    public RecyclerComicsListAdapter(List<ResultsItem> argDataset) {
```
Renommer le nom de la variable en paramètre en `listComics` pour comprendre ce quelle reçois

Merihel replied:
Plus simple, j'ai changé itemList en "comicList" et pareil pour argDataset (puis this.comicList = comicList dans le constructeur)
✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/adapter/RecyclerComicsListAdapter.java:
```diff
@@ -1,4 +1,59 @@
+import java.util.List;
+
+public class RecyclerComicsListAdapter extends RecyclerView.Adapter<ComicsListHolder> {
+    private List<ResultsItem> itemList;
```
Renommer la variable en `listComics` pour savoir qu'on à une list de comics

Merihel replied:
Changé en comicList (enlevé les s de partout du coup)
✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/adapter/RecyclerComicsListAdapter.java:
```diff
@@ -1,4 +1,59 @@
+        return itemList.size();
+    }
+
+    public String parseTimeStamp(String dateTimeStamp) {
```
Soit on externalise cette fonction en `utils` pour quelle puisse être utiliser dans les details aussi soit on fait une classe de `ComicsDataWrapper`

Merihel replied:
ComicDataWrapper ✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/adapter/RecyclerComicsListAdapter.java:
```diff
@@ -1,4 +1,59 @@
+
+    public String parseTimeStamp(String dateTimeStamp) {
+        Date d = new Date();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
```
Renommer le nom de la varaible parce que sdf me fais pas vraiment penser directement en SimpleDateFormat tu peux l'appeler par exemple `simpleDateFormat`

Merihel replied:
✔️

ronzierLoic commented on app/src/main/java/com/example/loic/comics_app_android/presentation/ui/viewholder/ComicsListHolder.java:
```diff
@@ -1,4 +1,22 @@
+
+import com.example.loic.comics_app_android.R;
+
+public class ComicsListHolder extends RecyclerView.ViewHolder {
```
Peut etre changer le nom de cette classe ou celle de l'activity car une fois il y a un s a comics une fois nan :/

Merihel replied:
✔️

Commit:
[FIX] Refactor variables and classes, add ComicDataWrapper -> date parse

- Fixed variables names, classes and files names
- Replaced parseTimestamp in Adapter by it's equivalent in new class ComicDataWrapper
